evaluator/src/main/kotlin/PackageRule.kt

```diff
@@ -70,6 +70,35 @@ open class PackageRule(
         licenseRules.forEach { it.evaluate() }
     }
 
+    /**
+     * A [RuleMatcher] that checks whether a severity of any vulnerability of the [package][pkg] using the specified
+     * [scoringSystem] is higher than [maxValue]. Tested using the [severityComparator]. Returns false if the severity
+     * is equal to [maxValue].
+     */
+    fun hasVulnerabilityWithSeverityOver(
```

Comment:
This API is rather basic. Is it planned to provide some abstractions over scoring systems? AFAIK, we already have a few classes to represent concrete scoring systems. So the mechanism to query vulnerabilities for a specific scoring system could be aligned to these classes (or vice versa). For instance, the scoring system classes could already provide a `Comparator`.

Reply:
As spoken on the phone:
I think we require this kind of low level API to support all scoring systems.
But for the known scoring systems (as of now `CVSS2` and `CVSS3`) I think it would be a nice addition to the API and I will create a follow up PR for this.